torch/nn/modules/activation.py

```diff
@@ -279,6 +279,36 @@ def __repr__(self):
         return self.__class__.__name__ + inplace_str
 
 
+class GLU(Module):
+    """Applies the gated linear unit function :math:`{GLU}(a, b)= a \otimes \sigma(b)`
+    where `a` is the first half of the input vector and `b` is the second half.
```

Comment:
I thought about this, but it's less convenient in practice. Gated linear units are used on the outputs of convolutional layers (https://arxiv.org/abs/1612.08083, https://arxiv.org/abs/1705.03122, https://arxiv.org/abs/1705.06824). You implement this with something like:

```
m = nn.Conv1d(input_planes, output_planes * 2)
F.glu(m(input))
```

For efficiency reasons, you don't want to split the Conv1d into two separate convolutions. If you took in two separate inputs, you'd have to add splitting code everywhere before you call glu.